commit 7e87d79de6d1eeeb47c8e43fb44962bce9c70664
Author: Dan Williams <dcbw@redhat.com>
Date:   Tue Aug 3 11:36:59 2010 -0700

    debian: don't require ifup for enabling loopback (bgo #625427)
    
    But use it if it's installed.

diff --git a/src/backends/NetworkManagerDebian.c b/src/backends/NetworkManagerDebian.c
index ef9475a..37eb645 100644
--- a/src/backends/NetworkManagerDebian.c
+++ b/src/backends/NetworkManagerDebian.c
@@ -1,3 +1,4 @@
+/* -*- Mode: C; tab-width: 4; indent-tabs-mode: t; c-basic-offset: 4 -*- */
 /* NetworkManager -- Network link manager
  *
  * Matthew Garrett <mjg59@srcf.ucam.org>
@@ -44,7 +45,11 @@
  */
 void nm_system_enable_loopback (void)
 {
-	nm_spawn_process ("/sbin/ifup lo");
+	/* ifupdown isn't always installed (bgo #625427) */
+	if (g_file_test ("/sbin/ifup", G_FILE_TEST_EXISTS))
+		nm_spawn_process ("/sbin/ifup lo");
+	else
+		nm_generic_enable_loopback ();
 }
 
 /*
